compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -241,16 +285,16 @@ fn struct_def_as_type(
 /// fn generics(self) -> [Quoted]
 fn struct_def_generics(
     interner: &NodeInterner,
-    mut arguments: Vec<(Value, Location)>,
+    arguments: Vec<(Value, Location)>,
     location: Location,
 ) -> IResult<Value> {
-    check_argument_count(1, &arguments, location)?;
+    let (argument, argument_location) = check_one_argument(arguments, location)?;
```

Comment:
This is the only reason why `check_one_argument` returns a tuple of (value, location). That's why I also kept all other helpers returning tuples of tuples... maybe it's fine.